src/Symfony/Component/Routing/Loader/AttributeClassLoader.php

```diff
@@ -144,7 +145,10 @@ public function load(mixed $class, string $type = null): RouteCollection
 
                 if (1 === $collection->count() - \count($routeNamesBefore)) {
                     $newRouteName = current(array_diff(array_keys($collection->all()), $routeNamesBefore));
-                    $collection->addAlias(sprintf('%s::%s', $class->name, $method->name), $newRouteName);
+                    try {
+                        $collection->addAlias(sprintf('%s::%s', $class->name, $method->name), $newRouteName);
```

Comment:
We could also assign everything to variables, compare and call `addAlias()` only when the name & alias are different.

Reply:
or we skip throwing in this case in addAlias?
I'm not sure this exception makes sense to me

Reply:
Yes, I agree but we'd change the behavior. Surely someone, somewhere, relies on this exception :sweat_smile:

Reply:
mmm :) 
then I'd prefer checking the value and not on the exception
one day we might throw for another reason exception

Reply:
Works for me!